src/languageServerInstaller.ts

```diff
@@ -90,27 +93,33 @@ export class LanguageServerInstaller {
 		});
 	}
 
-	installPkg(installDir: string, release, identifer: string, downloadUrl?: string): Promise<void> {
+	installPkg(installDir: string, release: { builds: any[]; version: string; }, userAgent: string): Promise<void> {
+		const destination: string = `${installDir}/terraform-ls_v${release.version}.zip`;
+
+		let platform = os.platform().toString();
+		if (platform === 'win32') {
+			platform = 'windows';
+		}
+		let arch = os.arch();
+		switch (arch) {
+			case 'x64':
+				arch = 'amd64'
+				break;
+			case 'x32':
+				arch = '386'
+				break;
+		}
+
+		const build = release.builds.find(b => b.os === platform && b.arch === arch);
+		const downloadUrl = build.url;
 		if (!downloadUrl) {
-			let arch = os.arch();
-			let platform = os.platform().toString();
-
-			switch (arch) {
-				case 'x64':
-					arch = 'amd64'
-					break;
-				case 'x32':
-					arch = '386'
-					break;
-			}
-			if (platform === 'win32') {
-				platform = 'windows'
-			}
-			downloadUrl = release.builds.find(b => b.os === platform && b.arch === arch).url;
-			if (!downloadUrl) {
-				// No matching build found
-				return Promise.reject();
-			}
+			// No matching build found
+			return Promise.reject();
+		}
+		try {
+			this.removeOldBinary(installDir, platform);
+		} catch {
+			// ignore
```

Comment:
no need to log?

Reply:
It's not an error we need to watch, it'll hit that if there's no previous binary in place (like with a new install).